collector/lib/SysdigService.cpp

```diff
@@ -192,6 +204,12 @@ void SysdigService::Start() {
 
   inspector_->start_capture();
 
+  // trigger the self check process only once capture has started,
+  // to verify the driver is working correctly. SelfCheckHandlers will
+  // verify the live events.
+  std::thread self_checks_thread(self_checks::start_self_check_process);
```

Comment:
Now I'm curious, can this introduce a concurrency, when the self-check is already started, but `SysdigService::Run` is not yet invoked?

Reply:
I think this is fine, because the inspector is definitely capturing events by the time we start the self-check process, so if anything there may be some buffering between here and a call to `SysdigService::Run` but we shouldn't lose any events.